bootstrap/bootstrap.go

```diff
@@ -75,49 +79,71 @@ func translateInterruptToCancel(ctx context.Context, wg *sync.WaitGroup, cancel
 func Run(
 	ctx context.Context,
 	cancel context.CancelFunc,
-	configDir, profileDir, configFileName string,
-	useRegistry bool,
+	configFileName string,
+	commonFlags commandline.CommonFlags,
 	serviceKey string,
-	config interfaces.Configuration,
+	serviceConfig interfaces.Configuration,
 	startupTimer startup.Timer,
 	dic *di.Container,
 	handlers []interfaces.BootstrapHandler) {
 
 	lc := logging.FactoryToStdout(serviceKey)
 	var err error
+	var configClient configuration.Client
 	var registryClient registry.Client
 	var wg sync.WaitGroup
 	translateInterruptToCancel(ctx, &wg, cancel)
 
 	// load configuration from file.
-	if err = configuration.LoadFromFile(configDir, profileDir, configFileName, config); err != nil {
+	if err = config.LoadFromFile(commonFlags.ConfigDirectory(), commonFlags.Profile(), configFileName, serviceConfig); err != nil {
 		fatalError(err, lc)
 	}
 
-	// override file-based configuration with environment variables.
-	bootstrapConfig := config.GetBootstrap()
-	registryInfo, startupInfo := configuration.OverrideFromEnvironment(bootstrapConfig.Registry, bootstrapConfig.Startup)
-	config.SetRegistryInfo(registryInfo)
+	// initialize config provider configuration for URL set in commandline options
+	providerConfig := types.ServiceConfig{}
```

Comment:
Yea, I struggled with the naming since `ConfigConfig` & `configSeviceConfig` just smelled bad... ;-)  How about `configProviderServiceConfig` for the variable name? Long but, actually read good. Or Suggestions?

Reply:
Yea, I struggled with the naming since ConfigConfig & configSeviceConfig just smelled bad... ;-) How about configProviderServiceConfig for the variable name? Long but, actually read good. Or Suggestions?

Reply:
configProviderInfo -- what you ended up changing it to -- seems fine.